### PR TITLE
Fix dt_day_of_week/day_of_year, str_cat/extract/partition/replace/rpartition

### PR DESCRIPTION
**Pull request by helmeleegy**
## What do these changes do?

- [ ] passes `flake8 modin`
- [ ] passes `black --check modin`
- [ ] signed commit with `git commit -s` 
- [ ] Resolves #? 
- [ ] tests added and passing

### CODE REVIEW

File: modin/pandas/series_utils.py [hunk 1]
```diff
@@ -343,11 +367,22 @@ def repeat(self, repeats):
         return Series(query_compiler=self._query_compiler.str_repeat(repeats))
 
     def rpartition(self, sep=" ", expand=True):
-        return Series(
-            query_compiler=self._query_compiler.str_rpartition(
-                sep=sep, expand=expand
+        if sep is not None and len(sep) == 0:
```
Comment (batur-ponder) [inline modin/pandas/series_utils.py:370]: can sep be None or should that raise ValueError as well ?

**Review by batur-ponder — APPROVED** [review]

**Review by naren-ponder — APPROVED** [review]

File: modin/pandas/series_utils.py (hunk 1 above)
Comment (pyrito) [reply modin/pandas/series_utils.py:370]: +1 to the question. it looks like if `sep` is none, it raises a `TypeError`. 

@helmeleegy can you double check that?

Comment (pyrito) [reply modin/pandas/series_utils.py:370]: In the spirit of trying to keep matching errors, we should try to get this to match.

Comment (helmeleegy) [reply modin/pandas/series_utils.py:370]: This was just putting back code that I believe was deleted by mistake in a previous PR. The same code still exists unchanged for str_partition.

**Pull request merged**